draft-ietf-quic-invariants.md

```diff
@@ -156,6 +174,29 @@ Example Structure {
 {: #fig-ex-format title="Example Format"}
 
 
+# Requirements for Intermediaries
+
+An intermediary MAY observe the QUIC version in certain packets, statefully note
+the version associated with the connection IDs (see {{connection-id}}) in the
+packet, and if it understands that version, parse and process packets with one
+of those connection IDs without regard for the invariants in this document.
+
+Intermediaries MUST implement a mode that, if they cannot associate a version
+with a packet's connection ID or do not understand the associated QUIC version,
+uses only the invariant fields described in this document to execute its
+function(s). This might mean that the intermediary cannot accomplish its
+function(s) at all, or that it implements a default behavior.
+
+Intermediaries MUST NOT implement a default behavior of dropping packets of QUIC
+versions they do not understand, or packets that have a connection ID they
+cannot associate with a QUIC version.
```

Comment:
You probably meant that intermediaries MUST NOT drop packets of unknown or unidentified version BECAUSE of the version being unknown or unidentified. The intermediaries should be able to (and will, regardless of what we say her) drop such packets for other reasons.

Reply:
The word "default" here is doing a lot of work, and I think it's communicating what you want.

I agree that people may violate this spec, even if they read it, though ignoring this provision is going to break both address migration and whatever version ossification mitigation we end up deploying.

Reply:
Perhaps "MUST NOT drop packets on the basis of..." would be avoid the wiggles of "default"?